bagit.py

```diff
@@ -632,6 +645,23 @@ def _validate_bagittxt(self):
             if first_line.startswith(codecs.BOM_UTF8):
                 raise BagValidationError("bagit.txt must not contain a byte-order mark")
 
+    def _path_is_dangerous(self, path):
+        """
+        Return true if path looks dangerous, i.e. potentially operates
+        outside the bagging directory structure, e.g. ~/.bashrc, ../../../secrets.json,
+            \\?\c:\, D:\sys32\cmd.exe
+        """
+        # Disallow dots, slashes, and tildes as first character
+        disallow = r'.\/~'
+        if path[0] in disallow:
+            return True
+        # Disallow drive letters at beginning of path, e.g c:, D:
+        drive_pat = re.compile(r'^[A-Za-z]:')
+        if drive_pat.match(path):
```

Comment:
Also, looking at the prose updates we've been adding to the spec – specifically https://github.com/loc-rdc/bagitspec/pull/1/commits/6673683d31473318b7d03f803b44336fc6f31319 – I think this would also want to have tests for `COM1:`, etc. Looking at https://msdn.microsoft.com/en-us/library/aa365247.aspx suggests we should also have some checks for the various reserved characters and the various tolerated path options on Windows (e.g. `C:/bad/but/not/uncommon/practice`)